src/rules/noUnboundMethodRule.ts

```diff
@@ -150,3 +209,16 @@ function isSafeUse(node: ts.Node): boolean {
             return false;
     }
 }
+
+function isWhitelisted(node: ts.Node, whitelist: string[]): boolean {
+    if (node.parent.kind === ts.SyntaxKind.TypeOfExpression) {
```

Comment:
When checking a node kind, use the `is*` methods from `tsutils`. It'll indicate to the type system whether the node is of the proper type, so you won't have to cast it later on.

For example, instead of `kind === ts.SyntaxKind.TypeOfExpression`, use `isTypeOfExpression`. It returns whether the given node `is ts.TypeOfExpression`.

Reply:
Using proper `is*` methods now